### ISSUE TITLE
vite:dep-pre-bundle cannot resolve entry for some Chakra UI packages

### ISSUE THREAD

**Issue by frle10** [chakra-ui/chakra-ui#6372]
### Description

I have a Vite SPA using React and Typescript template from Vite. After installing dependencies and running "yarn dev" which starts the development server, errors like this pop up: "[plugin vite:dep-pre-bundle] Failed to resolve entry for package "@chakra-ui/react-utils". The package may have incorrect main/module/exports specified in its package.json."

### Link to Reproduction

https://github.com/frle10/chakra-ui-package-versions-bug

### Steps to reproduce

1. Clone repo from link to reproduction
2. Install dependencies using yarn (yarn berry is set up in the repo, make sure you're installing with it)
3. Run yarn dev
4. See errors in logs from yarn dev that I described

### Chakra UI Version

2.2.4

### Browser

Google Chrome 103.0.5060.134

### Operating System

- [ ] macOS
- [ ] Windows
- [X] Linux

### Additional Information

Some error messages you should see:
```
[plugin vite:dep-scan] Failed to resolve entry for package "@chakra-ui/react".
The package may have incorrect main/module/exports specified in its package.json.
```
```
The plugin "vite:dep-scan" was triggered by this import
    src/main.tsx:4:31:
      4 │ import { ChakraProvider } from '@chakra-ui/react';
```

Interesting note, if you use yarn 1.22.19 and not yarn berry, everything works normally.

**Comment by thtliife** [comment]
This is because the latest published packages are all missing the dist folder. They are all like 2 or 3kb and empty except for readme and package.json

**Comment by carlreid** [comment]
Thanks for the resolutions list, was about to do that myself so you saved me some effort 🎊

@segunadebayo Is this somehow related to the move to pnpm? Possibly once https://github.com/chakra-ui/chakra-ui/pull/6361 is merged, all will work?

**Comment by thtliife** [comment]
> Thanks for the resolutions list, was about to do that myself so you saved me some effort 🎊
> 
> @segunadebayo Is this somehow related to the move to pnpm? Possibly once #6361 is merged, all will work?

@carlreid, version publish of [@chakra-ui/react](https://www.npmjs.com/package/@chakra-ui/react/v/2.2.5) appears to coincide with this gh action.... 
https://github.com/chakra-ui/chakra-ui/actions/runs/2729996327

triggered from pull #6369

**Comment by carlreid** [comment]
Yeh I saw that too, but it seems that PR I linked to might be preparing for a release due to the different build configuration

**Comment by thtliife** [comment]
I also think the reason yarn berry is seeing them as the latest versions, is because they are using a semantic version number, with no -dev or anything at the end... 
I'm pretty sure the version is how yarn determines whether or not the release is a production release... and not the tag.
So all of us with ^ versions in package.json (the default) will receive them as the latest patch version, as they are semantically versioned and public.

@frle10 , I think you will probably find you are running chakra-ui v2.2.5, and not v2.2.4

**Comment by carlreid** [comment]
That doesn't work either (pinning the version) on our side, as I had tried removing the `^` part but the resolver still tried to get spinner at `2.0.5`. My guess is that `alert` is using `^` as its own reference which yarn just picks up on. Though I really think if we were to pin a version in our own package.json, then the package underneath probably should respect that?

```
yarn why @chakra-ui/spinner
├─ @chakra-ui/alert@npm:2.0.4
│  └─ @chakra-ui/spinner@npm:2.0.5 (via npm:^2.0.4)
│
├─ @chakra-ui/alert@npm:2.0.4 [dddd1]
│  └─ @chakra-ui/spinner@npm:2.0.5 [c3602] (via npm:^2.0.4 [c3602])
│
├─ @chakra-ui/button@npm:2.0.4
│  └─ @chakra-ui/spinner@npm:2.0.4 (via npm:2.0.4)
│
├─ @chakra-ui/button@npm:2.0.4 [dddd1]
│  └─ @chakra-ui/spinner@npm:2.0.4 [dddd1] (via npm:2.0.4 [dddd1])
│
├─ @chakra-ui/react@npm:2.2.4
│  └─ @chakra-ui/spinner@npm:2.0.4 (via npm:2.0.4)
│
└─ @chakra-ui/react@npm:2.2.4 [2ab90]
   └─ @chakra-ui/spinner@npm:2.0.4 [dddd1] (via npm:2.0.4 [dddd1])
```

**Comment by thtliife** [comment]
> That doesn't work either (pinning the version) on our side, as I had tried removing the `^` part but the resolver still tried to get spinner at `2.0.5`. My guess is that `alert` is using `^` as its own reference which yarn just picks up on. Though I really think if we were to pin a version in our own package.json, then the package underneath probably should respect that?
> 
> 
> 
> ```
> 
> yarn why @chakra-ui/spinner
> 
> ├─ @chakra-ui/alert@npm:2.0.4
> 
> │  └─ @chakra-ui/spinner@npm:2.0.5 (via npm:^2.0.4)
> 
> │
> 
> ├─ @chakra-ui/alert@npm:2.0.4 [dddd1]
> 
> │  └─ @chakra-ui/spinner@npm:2.0.5 [c3602] (via npm:^2.0.4 [c3602])
> 
> │
> 
> ├─ @chakra-ui/button@npm:2.0.4
> 
> │  └─ @chakra-ui/spinner@npm:2.0.4 (via npm:2.0.4)
> 
> │
> 
> ├─ @chakra-ui/button@npm:2.0.4 [dddd1]
> 
> │  └─ @chakra-ui/spinner@npm:2.0.4 [dddd1] (via npm:2.0.4 [dddd1])
> 
> │
> 
> ├─ @chakra-ui/react@npm:2.2.4
> 
> │  └─ @chakra-ui/spinner@npm:2.0.4 (via npm:2.0.4)
> 
> │
> 
> └─ @chakra-ui/react@npm:2.2.4 [2ab90]
> 
>    └─ @chakra-ui/spinner@npm:2.0.4 [dddd1] (via npm:2.0.4 [dddd1])
> 
> ```

I had the same issue, that's why my workaround uses the resolutions key. 

It is designed to say, when searching for the key in resolutions, always resolve to the version in its value. 

So it works better than pinning versions, as all packages that request the resolutions ones, will resolve the version specified there

**Comment by frle10** [comment]
> This is because the latest published packages are all missing the dist folder. They are all like 2 or 3kb and empty except for readme and package.json For some reason, yarn 2+ is resolving them as the latest versions, even though they appear to have been published with the "dev" tag and not the "latest" one....
> 
> My temporary workaround is to address all the affected packages by resolving them to the last patch version with the "resolutions" key at the root of my package.json like below:
> 
> ```json
> {
>   "name": "project-name",
>   "version": "1.1.1",
>   "deps": {
>     "some-deps": "^1.0.0",
>   },
>   "resolutions": {
>     "@chakra-ui/accordion": "2.0.6",
>     "@chakra-ui/alert": "2.0.4",
>     "@chakra-ui/anatomy": "2.0.1",
>     "@chakra-ui/avatar": "2.0.5",
>     "@chakra-ui/breadcrumb": "2.0.4",
>     "@chakra-ui/button": "2.0.4",
>     "@chakra-ui/checkbox": "2.1.3",
>     "@chakra-ui/clickable": "2.0.4",
>     "@chakra-ui/close-button": "2.0.4",
>     "@chakra-ui/color-mode": "2.1.2",
>     "@chakra-ui/control-box": "2.0.4",
>     "@chakra-ui/counter": "2.0.4",
>     "@chakra-ui/css-reset": "2.0.1",
>     "@chakra-ui/descendant": "3.0.3",
>     "@chakra-ui/editable": "2.0.4",
>     "@chakra-ui/focus-lock": "2.0.5",
>     "@chakra-ui/form-control": "2.0.4",
>     "@chakra-ui/hooks": "2.0.4",
>     "@chakra-ui/icon": "3.0.4",
>     "@chakra-ui/icons": "2.0.4",
>     "@chakra-ui/image": "2.0.5",
>     "@chakra-ui/input": "2.0.4",
>     "@chakra-ui/layout": "2.1.1",
>     "@chakra-ui/live-region": "2.0.4",
>     "@chakra-ui/media-query": "3.2.0",
>     "@chakra-ui/menu": "2.0.6",
>     "@chakra-ui/modal": "2.1.2",
>     "@chakra-ui/number-input": "2.0.4",
>     "@chakra-ui/pin-input": "2.0.6",
>     "@chakra-ui/popover": "2.0.4",
>     "@chakra-ui/popper": "3.0.2",
>     "@chakra-ui/portal": "2.0.4",
>     "@chakra-ui/progress": "2.0.5",
>     "@chakra-ui/provider": "2.0.9",
>     "@chakra-ui/radio": "2.0.5",
>     "@chakra-ui/react-env": "2.0.4",
>     "@chakra-ui/react": "2.2.4",
>     "@chakra-ui/react-utils": "2.0.1",
>     "@chakra-ui/select": "2.0.4",
>     "@chakra-ui/skeleton": "2.0.9",
>     "@chakra-ui/slider": "2.0.4",
>     "@chakra-ui/spinner": "2.0.4",
>     "@chakra-ui/stat": "2.0.4",
>     "@chakra-ui/styled-system": "2.2.3",
>     "@chakra-ui/switch": "2.0.6",
>     "@chakra-ui/system": "2.2.2",
>     "@chakra-ui/table": "2.0.4",
>     "@chakra-ui/tabs": "2.0.6",
>     "@chakra-ui/tag": "2.0.4",
>     "@chakra-ui/textarea": "2.0.5",
>     "@chakra-ui/theme": "2.1.3",
>     "@chakra-ui/theme-tools": "2.0.5",
>     "@chakra-ui/toast": "3.0.2",
>     "@chakra-ui/tooltip": "2.0.5",
>     "@chakra-ui/transition": "2.0.4",
>     "@chakra-ui/utils": "2.0.4",
>     "@chakra-ui/visually-hidden": "2.0.4"
>   }
> }
> ```
> 
> @frle10 feel free to copy paste that resolutions key, and then do `yarn install` in your project root... should get you going in the meantime until the bad versions are fixed.

Thanks, I might use this or just downgrade to yarn 1.22.19 until this is fixed.

**Comment by Pagebakers** [comment]
This issue is really blocking, any news?

**Comment by frle10** [comment]
> This issue is really blocking, any news?

Agreed, it makes Chakra as a dependency unusable without workarounds on yarn berry. That isn't a small issue. But there was no activity for the past week on the repo :(

**Comment by segunadebayo** [comment]
I'm taking a look at this today and will get it fixed.

**Comment by segunadebayo** [comment]
Thanks for reporting this issue. I've fixed it in the latest release (2.2.6).

Please update to the most recent version to resolve the problem. If the issue persists, I'll re-open this issue.

Cheers

**Comment by nappalm** [comment]
![image](https://user-images.githubusercontent.com/35016608/196824993-525f1e37-afc2-4079-8a19-204f49c745eb.png)

I am having this problem when using pnpm
